db/compaction/compaction_iterator.h

```diff
@@ -433,8 +435,9 @@ class CompactionIterator {
   // True if the iterator has already returned a record for the current key.
   bool has_outputted_key_ = false;
 
-  // truncated the value of the next key and output it without applying any
-  // compaction rules.  This is used for outputting a put after a single delete.
+  // Truncate the value of the next key and output it without applying any
+  // compaction rules. This is an optimization for outputting a put after
+  // a single delete.
```

Comment:
Just to clarify a bit more on why to output a put after a single delete instead of dropping them together, "....This is an optimization for outputting a put after **outputting a single delete because there's an earlier snapshot preventing the put and single delete to be dropped together**. ""